app/models/sms/answer_parser.rb

```diff
@@ -0,0 +1,273 @@
+class Sms::AnswerParser
+  class ParseError < Sms::Error
+    attr_reader :type, :params
+
+    def initialize(type, params = {})
+      super(type)
+      @type = type
+      @params = params
+    end
+
+    def to_s
+      super + " #{@params.inspect}"
+    end
+  end
+
+  class Answer < Struct.new(:rank, :value)
```

Comment:
Let's talk about this one. Not sure what the intent here is but it's a bit unconventional for us to include a struct inside an outer class like this.